core/src/main/scala/format/BinaryFen.scala

```diff
@@ -4,172 +4,175 @@ package format
 import chess.bitboard.{ Bitboard, Board as BBoard }
 import chess.variant.*
 
-import scala.collection.mutable.ListBuffer
+import scala.collection.mutable.ArrayBuilder
+
+case class BinaryFen(value: Array[Byte]):
```

Comment:
```suggestion
case class BinaryFen(value: Array[Byte]) extends AnyVal:
```

We can extends `AnyVal` to make it value class, similar to opaque type. But with this approach We secure the `equals` some how.